### PR TITLE
Using ami_name and ami_suffix when building the imageName

### PR DESCRIPTION
**Pull request by kalymero**

### CODE REVIEW

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/util/DefaultImageNameFactory.groovy [hunk 1]
```diff
@@ -75,10 +75,19 @@ public class DefaultImageNameFactory implements ImageNameFactory {
     // We need to replace the original fully-qualified package name with the unqualified package name before using
     // it in the target image name.
     def baseImagePackageName = osPackageName?.name ?: firstPackageName
-    def imageName = baseImagePackageName ? "$baseImagePackageName-" : ""
+
```
Comment (duftler) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/util/DefaultImageNameFactory.groovy]: Looks like line 77 (the `baseImagePackageName` declaration/assignment) and its comment can move inside of the `else` clause.

File: rosco-core/src/test/groovy/com/netflix/spinnaker/rosco/providers/util/DefaultImageNameFactorySpec.groovy [hunk 2]
```diff
@@ -63,6 +63,64 @@ class DefaultImageNameFactorySpec extends Specification {
       packagesParameter == "nflx-djangobase-enhanced=0.1-3"
   }
 
+  void "should build the imageName based on the ami_name tag"() {
```
Comment (duftler) [inline rosco-core/src/test/groovy/com/netflix/spinnaker/rosco/providers/util/DefaultImageNameFactorySpec.groovy]: Nice thorough tests.
Please try to match the formatting of the neighboring tests (the content of the `setup/when/then` clauses is indented 2 spaces).

File: rosco-web/config/packer/aws-ebs.json [hunk 3]
```diff
@@ -32,7 +32,11 @@
     "ami_name": "{{user `aws_target_ami`}}",
     "associate_public_ip_address": "{{user `aws_associate_public_ip_address`}}",
     "enhanced_networking": "{{user `aws_enhanced_networking`}}",
-    "tags": {"appversion": "{{user `appversion`}}", "build_host": "{{user `build_host`}}"},
+    "tags": {
+      "appversion": "{{user `appversion`}}",
+      "build_host": "{{user `build_host`}}",
+      "Name": "{{user `aws_target_ami`}}"
```
Comment (duftler) [inline rosco-web/config/packer/aws-ebs.json]: What is this new tag used for? Can it be lowercased like the others?

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/util/DefaultImageNameFactory.groovy (hunk 1 above)
Comment (kalymero) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/util/DefaultImageNameFactory.groovy]: indeed, moved

File: rosco-core/src/test/groovy/com/netflix/spinnaker/rosco/providers/util/DefaultImageNameFactorySpec.groovy (hunk 2 above)
Comment (kalymero) [inline rosco-core/src/test/groovy/com/netflix/spinnaker/rosco/providers/util/DefaultImageNameFactorySpec.groovy]: ouch

File: rosco-web/config/packer/aws-ebs.json (hunk 3 above)
Comment (kalymero) [inline rosco-web/config/packer/aws-ebs.json]: nevermind, it's a leak from another template

**Comment by duftler** [comment]
LGTM. Please squash the commits and I'll merge.

**Pull request merged**